From: Don Dutile <ddutile@redhat.com>
Date: Fri, 24 Apr 2009 14:44:34 -0400
Subject: [misc] VT-d: add DMAR-related timeout definition
Message-id: 49F20892.1090501@redhat.com
O-Subject: Re: [RHEL5.4 PATCH 4/6] VT-d for KVM: Add DMAR-related timeout definition
Bugzilla: 480411
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Chris Wright <chrisw@redhat.com>

BZ 480411

In upstream, this DMAR-related timeout
is defined in arch/x86/include/asm/iommu.h .

It's about the only thing in that file, and
it didn't exist in RHEL5, so placed it in
include/asm-x86_64/proto.h  for simplicity.

Please review & ack.

- Don

>From 6fc1c8e526dc0ce9b4e93976604fedd09654125e Mon Sep 17 00:00:00 2001
From: Mark McLoughlin <markmc@redhat.com>
Date: Mon, 17 Nov 2008 19:45:35 +0000
Subject: [PATCH 4/6] vt-d: add DMAR_OPERATION_TIMEOUT definition

>From upstream asm/iommu.h - both tsc_khz and other arch specific
iommu related bits are definied in asm/proto.h on 2.6.18, so it
makes most sense to add it there.

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

diff --git a/include/asm-x86_64/proto.h b/include/asm-x86_64/proto.h
index c448d24..e3be7a7 100644
--- a/include/asm-x86_64/proto.h
+++ b/include/asm-x86_64/proto.h
@@ -114,6 +114,10 @@ extern void pci_iommu_shutdown(void);
 extern void no_iommu_init(void);
 extern int force_iommu, no_iommu;
 extern int iommu_detected;
+
+/* 10 seconds */
+#define DMAR_OPERATION_TIMEOUT ((cycles_t) tsc_khz*10*1000)
+
 #ifdef CONFIG_IOMMU
 extern void gart_iommu_init(void);
 extern void gart_iommu_shutdown(void);
